### PR TITLE
Add configure with flags

### PR DESCRIPTION
**Pull request by jaeklim**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Adding distribute flags and configure method to stop getting a new release when it is disabled.

## Related PRs or issues

[AB#75289](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/75289)

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  XCTAssertEqual(MSDistributeFlagsDisableAutomaticCheckForUpdate, distribute.distributeFlags);
+}
+
+- (void)testCheckForUpdate {
+
+  // If
+  NSString *updateToken = @"UpdateToken";
+  NSString *distributionGroupId = @"DistributionGroupId";
+  NSString *releaseHash = @"ReleaseHash";
+  id distributeMock = OCMPartialMock([MSDistribute new]);
```
Comment (MatkovIvan) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m]: Call stopMocking for partial mocks explicitly

File: AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h [hunk 2]
```diff
@@ -171,6 +181,23 @@ static NSString *const kMSTesterAppUpdateSetupFailedKey = @"MSTesterAppUpdateSet
 - (nullable NSURL *)buildTokenRequestURLWithAppSecret:(NSString *)appSecret
                                           releaseHash:(NSString *)releaseHash
                                           isTesterApp:(BOOL)isTesterApp;
+/**
+ * Configure Distribute options before the service starts.
```
Comment (MatkovIvan) [inline AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h]:
```suggestion
 * Configure distribute options before the service starts.
```

Comment (jaeklim) [reply AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h]: This is a name of the module so it started with an uppercase.

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+@interface MSDistributeConfigureTests : XCTestCase
+
+@end
+
+@implementation MSDistributeConfigureTests
+
+- (void)testConfigureFlagsBeforeStart {
+
+  // If
+  [MSDistribute sharedInstance].distributeFlags = -1;
```
Comment (dhei) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m]: Why setting `distributeFlags` to an invalid value like `-1`? 

`XCTAssertEqual(MSDistributeFlagsDisableAutomaticCheckForUpdate, [MSDistribute sharedInstance].distributeFlags);`

I don't understand how this test would work, `1 ^ -1` should result in `-2` not `1`.

Comment (jaeklim) [reply AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m]: This test is to make sure the flag got changed by the API call. `-1` won't be used but it just want to make sure of the value change only in unit test.

Comment (jaeklim) [reply AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m]: > I don't understand how this test would work, 1 ^ -1 should result in -2 not 1.
There is no this kind of operations during the test.

Comment (dhei) [reply AppCenterDistribute/AppCenterDistributeTests/MSDistributeConfigureTests.m]: You are right. Sorry for the confusion, I misunderstood the code.

**Pull request merged**